File: client/src/ee/utils/tier.ts
import { Tier } from "@/ee/types/types";

interface FeatureSettings {
  users: number;
  managedGenerations: boolean;
  diffTooling: boolean;
  annotations: boolean;
  vibeSynthesis: boolean;
}

// Define the type for the features object
type FeaturesConfig = {
  [key in Tier]: FeatureSettings;
};

export const features: FeaturesConfig = {
  [Tier.FREE]: {
    users: 1,
    managedGenerations: false,
    diffTooling: false,
    annotations: false,
    vibeSynthesis: false,
  },
  [Tier.PRO]: {
    users: 5,
    managedGenerations: true,
    diffTooling: true,
    annotations: true,
    vibeSynthesis: false,
  },
  [Tier.TEAM]: {
    users: Infinity,
    managedGenerations: true,
    diffTooling: true,
    annotations: true,
    vibeSynthesis: true,
  },
  [Tier.ENTERPRISE]: {
    users: Infinity,
    managedGenerations: true,
    diffTooling: true,
    annotations: true,
    vibeSynthesis: true,
  },
};

File: app/lilypad/server/_utils/tier.py
"""Utility functions for tier management to avoid circular imports."""

import logging

from ee import Tier

from ...ee.server.features import cloud_features

logger = logging.getLogger(__name__)


def get_display_retention_days(tier: Tier) -> int | None:
    """Get display retention days for a tier.

    This defines the display retention policy:
    - FREE: 30 days (though data is stored for 90 days)
    - PRO: 180 days
    - TEAM/ENTERPRISE: Unlimited

    Args:
        tier: The organization tier

    Returns:
        Number of days to retain display data, or None for unlimited
    """
    feature_settings = cloud_features.get(tier)
    if not feature_settings:
        return None  # pragma: no cover

    # Convert float('inf') to None for unlimited retention
    days = feature_settings.display_retention_days
    return None if days == float("inf") else int(days)
